### PR TITLE
Feature/songs

### PR DESCRIPTION
**Pull request by miggelucas**
## What I Have done:

* Song Model
* SongsView and ViewModel for displaying registered Songs
* Create Song View and ViewModel for creating new songs
    * User can label a name, artist and append pedals
    * User can adjust level for every pedal knob selected
    * If label and name are missing an alert will be presented
* ViewModel testing 
* Auxiliar SelectPedalView for CreateSongView
    * LightSelectorView for indicate selected pedals for the song

### CODE REVIEW

**Comment by miggelucas** [comment]
Please, do not merge yet. Still working on improvements. Also, this branch has commits from other PR (feature/PedalRowSwipeActions), so wait until its done before merging this one.

**Comment by miggelucas** [comment]
@matheusberger @matheusmigge 
All set. Hope that everything is working smoothly.
Please look up for bugs and feel free to propose any alteration.

**Review by matheusmigge — CHANGES_REQUESTED** [review]
I've done some testing and I wrote some suggestions:

- On old versions of Xcode, all computed variable must be declared with a "return". Could you correct this in the variables where the return is missing?
- In CreateSongView, the alert when the label and name are missing is not showing up. Could you check?

File: PedalCore/ViewModels/SongsViewModel.swift
Comment (matheusmigge) [inline PedalCore/ViewModels/SongsViewModel.swift]: Please add a return in this computed variable

File: PedalCore/ViewModels/SongsViewModel.swift
Comment (matheusmigge) [inline PedalCore/ViewModels/SongsViewModel.swift]: And also here

Comment (miggelucas) [reply PedalCore/ViewModels/SongsViewModel.swift]: done

**Comment by miggelucas** [comment]
@matheusmigge  Please, could you try to pull the last commit and verify if the problems still persist

File: PedalCore/Views/Songs/SelectPedalView.swift
Comment (matheusmigge) [inline PedalCore/Views/Songs/SelectPedalView.swift]:
```suggestion
                            withAnimation {
```
This change made the code works on older Xcode versions! Is it ok to remove "(.smooth)" with any penalty for our app?

Comment (miggelucas) [reply PedalCore/Views/Songs/SelectPedalView.swift]: done!

File: PedalCore/Views/Songs/CreateSongView.swift
Comment (matheusmigge) [inline PedalCore/Views/Songs/CreateSongView.swift]:
```suggestion
                    TextField("Band name", text: $viewModel.bandName, prompt: Text("Artist name"))
```

File: PedalCore/Views/Songs/CreateSongView.swift
Comment (matheusmigge) [inline PedalCore/Views/Songs/CreateSongView.swift]:
```suggestion
                    TextField("Song name", text: $viewModel.songName, prompt: Text("Song name"))
```

**Review by matheusmigge — CHANGES_REQUESTED** [review]
Thank you for your changes! While testing, i've noticed some other things:

- When adding a new song, it is not possible to create a new pedal. It could be a real experience, while creating the new song and selecting the pedals, the user could notice that a pedal for the song is missing. It would be interesting to be possible to create a new pedal in this stage of the experience. What do you think?
- I've done some suggestions on the texts showed for the user, simple changes. Could you check if you agree?
- Also noticed that the CreateSongView .sheet could have something like a navigationTitle, like "Create song". It feels like there is a missing element on top of the screen:
image

I think that will be all  🤗

**Comment by miggelucas** [comment]
> Thank you for your changes! While testing, i've noticed some other things:
> 
> * When adding a new song, it is not possible to create a new pedal. It could be a real experience, while creating the new song and selecting the pedals, the user could notice that a pedal for the song is missing. It would be interesting to be possible to create a new pedal in this stage of the experience. What do you think?
> * I've done some suggestions on the texts showed for the user, simple changes. Could you check if you agree?
> * Also noticed that the CreateSongView .sheet could have something like a navigationTitle, like "Create song". It feels like there is a missing element on top of the screen:
> 
> image
> I think that will be all 🤗


Hi matheus, thanks for the testing and giving suggestion. Gonna reply to them in order:

1. Was thinking about it and understood the create pedal feature would make sense to be able to have an access point when the user were creating a song (like noticing a pedal is missing and making a new pedal register easily). However, I also thinking this should not be the main entry for registering new Pedals, and since by now we don't have one (we are still trying to figure out how to attach both features properly), I would take your suggestion with caution so don't bias our future steps and mislead the app architecture development. I believe it would be safer to wait until a layer of source of user data beside the viewModels to be done first. Never the less, if you still have reasons to try adding the feature right away, I can do it in other PR since this one seems big enough already;
2. Text Labels were both great so already made a commit from suggestions;
3. Tried on simulator and could't reproduce your screenshot. Maybe it's something related about the outdated Xcode of you. However, seems a nice idea to give the view a proper title. Gonna work this on.


Sorry for long response

**Review by matheusberger — CHANGES_REQUESTED** [review]
Feature looks great, but please use the same "add" button standard from the PedalView (big button on the bottom of the screen)

**Comment by matheusmigge** [comment]
@miggelucas Thank you for your answer! 
I'm sorry but I forgot to mention in the last review an unusual behaviour when selecting a pedal for a song:

- The pedal list row is only clickable in the texts and in the green circle, not in the whole row area as I think it should. What do you think? 

image

**Review by matheusmigge — APPROVED** [review]
Good job @miggelucas!

**Review by matheusberger — APPROVED** [review]

**Pull request merged**
